internal/common_cluster_info/common_cluster_info.go

```diff
@@ -61,6 +57,33 @@ func GetInstallationID(ctx context.Context, client kubernetes.Interface, podNSNa
 	case "DaemonSet":
 		return string(podOwner[0].UID), nil
 	default:
-		return "", fmt.Errorf("expected pod owner reference to be ReplicaSet or DeamonSet, got %s", podOwner[0].Kind)
+		return podOwner[0].Kind, nil
+	}
+}
+
+// GetDeploymentName returns the name of the Deployment
+func GetDeploymentName(ctx context.Context, client kubernetes.Interface, podNSName types.NamespacedName) (string, error) {
+	pod, err := client.CoreV1().Pods(podNSName.Namespace).Get(ctx, podNSName.Name, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	owners := pod.GetOwnerReferences()
+	owner := owners[0]
+	switch owner.Kind {
+	case "ReplicaSet":
+		replicaSet, err := client.AppsV1().ReplicaSets(podNSName.Namespace).Get(ctx, owner.Name, metav1.GetOptions{})
+		if err != nil {
+			return "", err
+		}
+		for _, replicaSetOwner := range replicaSet.GetOwnerReferences() {
+			if replicaSetOwner.Kind == "Deployment" {
+				return replicaSetOwner.Name, nil
+			}
+		}
+		return "", fmt.Errorf("replicaset %s has no owner", replicaSet.Name)
+	case "DaemonSet":
+		return owner.Name, nil
+	default:
+		return owner.Kind, nil
```

Comment:
This might actually be better as owner.Name
```suggestion
		return owner.Name, nil
```